lang/java/avro/src/test/java/org/apache/avro/TestSchemaValidation.java

```diff
@@ -15,183 +15,316 @@
  * implied.  See the License for the specific language governing
```

Comment:
Although some are redundant, I revived the old tests as well as my new additions.
Like you say, this PR is big enough as it is - if we decide to clean up, it could be done in another PR